modal_images/token_identity.py

```diff
@@ -0,0 +1,155 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from modal import Secret, Stub, gpu, method
+
+from modal_images.gemma import image as gemma_image
+from modal_images.mistral import image as mistral_image
+from obvs.vis import create_heatmap, plot_surprisal
+
+images = {
+    "gemma2": gemma_image,
+    "gemma7": gemma_image,
+    "mistral": mistral_image,
+    "gpt2": gemma_image,
+}
+
+
+model_names = {
+    "llamatiny": "TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T",
+    "llama": "meta-llama/Llama-2-13b-hf",
+    "gpt2": "gpt2",
+    "mamba": "MrGonao/delphi-mamba-100k",
+    "mistral": "mistralai/Mistral-7B-v0.1",
+    "gptj": "EleutherAI/gpt-j-6B",
+    "gemma2": "google/gemma-2b",
+    "gemma7": "google/gemma-7b",
+}
+
+
+stub = Stub(
+    image=gemma_image,
+    name="token_identity",
+    secrets=[Secret.from_name("my-huggingface-secret")],
+)
+
+
+@stub.cls(
+    # gpu=gpu.A100(memory=40, count=1),
+    # gpu=gpu.A10G(count=1),   # 24 GB
+    gpu=gpu.T4(count=1),  # 16 GB
+    # cpu=1,
+    timeout=60 * 30,
+    container_idle_timeout=60 * 5,
+)
```

Comment:
different gpus have different costs, so if you are running gemma2 for instance you can go for a smaller one.

they also have H100s with 80GB if you want to run a mistral test and dont want to worry about running out of memory :)